src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.parser;
+
+import duke.command.DeleteCommand;
+import duke.command.DeadlineCommand;
+import duke.command.Command;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.ToDoCommand;
+import duke.command.FindCommand;
+import duke.exception.DukeException;
+import duke.shared.Messages;
+import java.util.Arrays;
+
+/**
+ * Parses command.
+ */
+public class Parser {
+    /**
+     * Converts the command into command object.
+     * @param command command inputted by the user
+     * @return converted command object
+     * @throws DukeException throws by {@link #findCommand(String[])}
+     * @throws NumberFormatException throws by {@link #findCommand(String[])}
+     */
+    public static Command parse(String command) throws DukeException, NumberFormatException {
+        String[] commandArr = command.split("\\s+");
+        return findCommand(commandArr);
+    }
+
+    /**
+     * Returns the correct command object based on user inputted command.
+     * @param commands is the last command entered by the user
+     * @throws DukeException if user inputted an invalid command
+     * @throws NumberFormatException if user inputted an invalid description for done and delete command
+     */
+    public static Command findCommand(String[] commands) throws DukeException, NumberFormatException {

```

Comment:
Can consider shortening the conditional statement with these by grouping the tasks under one condition so that you don't need to keep on throwing 
```
 else if (commands.length == 1) {
                throw new DukeException(String.format(Messages.DESCRIPTION_MISSING_EXCEPTION, "done"));
            } else {
                throw new DukeException(Messages.UNKNOWN_COMMAND_EXCEPTION);
            }
```
This will reduce the chances of human error and increase readability